packages/actor-rdf-resolve-quad-pattern-string-source/lib/ActorRdfResolveQuadPatternStringSource.ts

```diff
@@ -0,0 +1,81 @@
+import type { MediatorRdfParseHandle } from '@comunica/bus-rdf-parse';
+import { getContextSource
+  , ActorRdfResolveQuadPattern } from '@comunica/bus-rdf-resolve-quad-pattern';
+import type {
+  IActionRdfResolveQuadPattern, IActorRdfResolveQuadPatternArgs, IActorRdfResolveQuadPatternOutput,
+} from '@comunica/bus-rdf-resolve-quad-pattern';
+import type { IActorTest } from '@comunica/core';
+import type { ISerializeDataSource } from '@comunica/types';
+import type * as RDF from '@rdfjs/types';
+import type { AsyncIterator } from 'asynciterator';
+import { wrap } from 'asynciterator';
+import { Readable } from 'readable-stream';
+
+/**
+ * A comunica RDF Resolve Quad Pattern String Source RDF Resolve Quad Pattern Actor.
+ */
+export class ActorRdfResolveQuadPatternStringSource extends ActorRdfResolveQuadPattern {
+  private readonly sourceType = 'stringSource';
+  private readonly mediatorRdfParse: MediatorRdfParseHandle;
+
+  public constructor(args: IActorRdfResolveQuadPatternStringSource) {
+    super(args);
+    this.mediatorRdfParse = args.mediatorRdfParse;
```

Comment:
Rather than creating a `string-source` quad-pattern-resolver; I please ask that you create a `string` actor on top of the `dereference` bus (similar to https://github.com/comunica/comunica/blob/master/packages/actor-dereference-file/lib/ActorDereferenceFile.ts).

This should allow you to significantly reduce the amount of code - and also means that this solution will also immediately apply to other data sources for things like shapes and rules in the `link-traversal` and `reasoning` repositories.

Reply:
Strings are not really something that you can _dereference_ though, so not sure I fully agree with this.